### PR TITLE
Added Provider: DreamAnime

### PR DESCRIPTION
**Pull request by AbdullahM0hamed**

### CODE REVIEW

**Comment by uali6981** [comment]
Add `chosing of server` in config.json file. This help to choose server between `trollvid` and `mp4upload`.
Thanks

**Comment by AbdullahM0hamed** [comment]
@uali6981, done. You can now configure the server, and the version (whether it should download subbed or dubbed).

**Comment by uali6981** [comment]
After updating I got this error
Screenshot_20200330-120136


and here `DEBUG` for this

Screenshot_20200330-120856
Screenshot_20200330-120903
Screenshot_20200330-120910
Screenshot_20200330-120920
Screenshot_20200330-120933
Screenshot_20200330-120942
Screenshot_20200330-120946

File: anime_downloader/sites/dreamanime.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        subbed = []
+        dubbed = []
+        _all = soup.find_all("div", {"class":"episode-wrap"})
+        for i in _all:
+            ep_type = i.find("div", {"class":re.compile("ep-type type-.* dscd")}).text
+            if ep_type == 'Sub':
+                subbed.append(i.find("a").get("data-src"))
+            elif ep_type == 'Dub':
+                dubbed.append(i.find("a").get("href"))
+        return eval(version)
```
Comment (vn-ki) [inline anime_downloader/sites/dreamanime.py]: Why are using `eval`?

**Comment by AbdullahM0hamed** [comment]
@vn-ki I'm using Eval so the variable `dubbed` is returned is config has version set to dub, and sub if it's set to `dub`.

@Blatzar Had it like that initially, made an oversight when cleaning the code a bit. Fixed.

File: anime_downloader/sites/dreamanime.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    server: One of ['mp4upload', 'trollvid']
+        Selects the server to download from.
+    """
+
+    sitename='dreamanime'
+    url = f'https://{sitename}.fun/search'
+
+    @classmethod
+    def search(cls, query):
+        results = helpers.get("https://dreamanime.fun/search?term=" + quote_plus(query)).text
```
Comment (ngomile) [inline anime_downloader/sites/dreamanime.py]: Unnecessary hardcoding of url parameters, search_url is defined on line 20 as url but manually inputted here. Should use url and pass parameters as such helpers.get('https://dreamanime.fun/search', params={'term': 'naruto'}).text and remove import of quote_plus.

**Pull request merged**